federation/pkg/federation-controller/service/dns.go

```diff
@@ -205,7 +205,7 @@ func (s *ServiceController) ensureDnsRrsets(dnsZoneName, dnsName string, endpoin
 			// Need an appropriate CNAME record.  Check that we have it.
 			newRrset := rrsets.New(dnsName, []string{uplevelCname}, minDnsTtl, rrstype.CNAME)
 			glog.V(4).Infof("No healthy endpoints for %s.  Have recordset %v. Need recordset %v", dnsName, rrset, newRrset)
-			if rrset == newRrset {
+			if dnsprovider.ResourceRecordSetsEquivalent(rrset, newRrset) {
 				// The existing rrset is equal to the required one - our work is done here
 				glog.V(4).Infof("Existing recordset %v is equal to needed recordset %v, our work is done here.", rrset, newRrset)
```

Comment:
super nit: if you're going to change the other message from equal to equivalent ...